src/Orchard.Web/Modules/Orchard.Autoroute/Drivers/AutoroutePartDriver.cs

```diff
@@ -20,54 +25,99 @@ public class AutoroutePartDriver : ContentPartDriver<AutoroutePart> {
+
+            //if we are editing an existing content item
+            if (part.Record.Id != 0) {
+                ContentItem contentItem = _contentManager.Get(part.Record.ContentItemRecord.Id);
+                var aspect = contentItem.As<ILocalizableAspect>();
+
+                if (aspect != null) {
+                    itemCulture = aspect.Culture;
+                }
+            }
+
+            if (settings.UseCulturePattern) {
+                //if we are creating from a form post we use the form value for culture
+                HttpContextBase context = _httpContextAccessor.Current();
+                if (!String.IsNullOrEmpty(context.Request.Form["Localization.SelectedCulture"])) {
+                    itemCulture = context.Request.Form["Localization.SelectedCulture"].ToString();
+                }
+            }
+
+            //we update the settings assuming that when 
+            //pattern culture = null or "" it means culture = default website culture
+            //for patterns that we migrated
+            foreach (RoutePattern pattern in settings.Patterns.Where(x => String.IsNullOrWhiteSpace(x.Culture))) {
+                pattern.Culture = _cultureManager.GetSiteCulture(); ;
+            }
+
+            //we do the same for default patterns
+            foreach (DefaultPattern pattern in settings.DefaultPatterns.Where(x => String.IsNullOrWhiteSpace(x.Culture))) {
+                pattern.Culture = _cultureManager.GetSiteCulture();
+            }
+
             // if the content type has no pattern for autoroute, then use a default one
-            if(!settings.Patterns.Any()) {
-                settings.AllowCustomPattern = true;
-                settings.AutomaticAdjustmentOnEdit = false;
-                settings.DefaultPatternIndex = 0;
-                settings.Patterns = new List<RoutePattern> {new RoutePattern {Name = "Title", Description = "my-title", Pattern = "{Content.Slug}"}};
+            if (!settings.Patterns.Any(x => String.Equals(x.Culture, itemCulture, StringComparison.OrdinalIgnoreCase))) {
+                settings.Patterns = new List<RoutePattern> { new RoutePattern { Name = "Title", Description = "my-title", Pattern = "{Content.Slug}", Culture = itemCulture } };
+            }
 
-                _notifier.Warning(T("No route patterns are currently defined for this Content Type. If you don't set one in the settings, a default one will be used."));
+            // if the content type has no defaultPattern for autoroute, then use a default one
+            if (!settings.DefaultPatterns.Any(x => String.Equals(x.Culture, itemCulture, StringComparison.OrdinalIgnoreCase))) {
+                //if we are in the default culture check the old setting
+                if (String.Equals(itemCulture, _cultureManager.GetSiteCulture(), StringComparison.OrdinalIgnoreCase)) {
+                    if (!String.IsNullOrEmpty(part.TypePartDefinition.Settings["AutorouteSettings.DefaultPatternIndex"])) {

```

Comment:
I refactored this bit because it was breaking the autoroute part feature for new Orchard installations, so you may want to get latest from dev and update your PR, sorry.